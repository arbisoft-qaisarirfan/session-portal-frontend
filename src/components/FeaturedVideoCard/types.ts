export type FeaturedVideoCardProps = {
  isVisible?: boolean;
  className?: string;
  event_time: string;
  thumbnail?: string;
  title: string;
  description: string;
  video_duration: number;
  workstream_id: string;
  width?: string;
};
