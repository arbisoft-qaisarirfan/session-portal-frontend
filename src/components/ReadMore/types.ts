export interface ReadMoreProps {
  amountOfWords?: number;
  className?: string;
  showLessText: string;
  showMoreText: string;
  text: string;
}
