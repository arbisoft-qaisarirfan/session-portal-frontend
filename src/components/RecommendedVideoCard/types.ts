export interface RecommendedVideoCardProps {
  className?: string;
  date: string;
  duration: string;
  imgUrl: string;
  onClick?: VoidFunction;
  organizer?: string;
  title: string;
  width?: string;
}
