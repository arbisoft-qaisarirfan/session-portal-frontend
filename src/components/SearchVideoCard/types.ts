export type SearchVideoCardProps = {
  id: number;
  className?: string;
  event_time: string;
  thumbnail?: string;
  title: string;
  description: string;
  video_duration: number;
  workstream_id: string;
  width?: string;
};
