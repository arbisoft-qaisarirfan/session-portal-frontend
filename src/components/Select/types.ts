import { SelectProps as MuiSelectProps, SelectChangeEvent } from "@mui/material/Select";

export type SelectProps = {
  menuItems: { value: string; label: string }[];
  handleChange: (event: SelectChangeEvent) => void;
} & MuiSelectProps;
