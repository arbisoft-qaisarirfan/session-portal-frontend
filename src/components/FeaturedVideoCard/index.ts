import FeaturedVideoCard from "./featuredVideoCard";

export default FeaturedVideoCard;
