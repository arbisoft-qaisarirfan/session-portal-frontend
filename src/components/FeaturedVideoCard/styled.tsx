import Box from "@mui/material/Box";
import Card from "@mui/material/Card";
import { cardContentClasses } from "@mui/material/CardContent";
import { styled, css } from "@mui/material/styles";
import { typographyClasses } from "@mui/material/Typography";

import { shouldForwardProp } from "@/utils/styleUtils";

export const FeaturedVideoCardContainer = styled(Card, {
  name: "FeaturedVideoCardContainer",
  shouldForwardProp,
})<{ $width: string }>(({ theme, $width }) => {
  return css`
    display: flex;
    background-color: rgba(255, 255, 255, 0.06);
    border-radius: 12px;
    padding: 30px;
    border-radius: ${theme.shape.borderRadius + 8}px;
    overflow: hidden;
    width: ${$width};
    margin-top: 10px;

    .${cardContentClasses.root} {
      align-items: flex-start;
      display: flex;
      flex-direction: row;
      gap: ${theme.spacing(2.5)};
      padding: 0;
      :last-child {
        padding: 0;
      }

      img {
        border-radius: ${theme.shape.borderRadius + 8}px;
        height: auto;
        object-fit: cover;
        width: 400px;
      }

      .video-detail {
        width: calc(100% - 400px);
        display: flex;
        flex-direction: column;
        gap: 7px;

        .${typographyClasses.h3} {
          -webkit-box-orient: vertical;
          -webkit-line-clamp: 2;
          color: ${theme.palette.colors.white};
          display: -webkit-box;
          font-size: 28px;
          font-style: normal;
          font-weight: 500;
          letter-spacing: 0.4px;
          line-height: 35px;
          overflow: hidden;
          text-overflow: ellipsis;
          padding-bottom: 10px;
        }

        .date-time,
        .organizer-name {
          color: ${theme.palette.colors.gray};
          font-size: 18px;
          font-style: normal;
          font-weight: 500;
          line-height: 17px;
          letter-spacing: 0.4px;
        }

        .video-description {
          color: ${theme.palette.colors.white};
          font-size: 18px;
          font-style: normal;
          font-weight: 500;
          line-height: 21px;
          letter-spacing: 0.4px;
          padding-top: 10px;
        }
      }
    }

    ${theme.breakpoints.down("xl")} {
      .${cardContentClasses.root} {
        .video-detail {
          .${typographyClasses.h3} {
            font-size: 20px;
            line-height: 24px;
          }
          .date-time,
          .organizer-name {
            font-size: 14px;
            line-height: 14px;
          }
          .video-description {
            font-size: 14px;
            line-height: 14px;
          }
        }
      }
    }

    ${theme.breakpoints.down("lg")} {
      padding: ${theme.spacing(1)};
      .${cardContentClasses.root} {
        flex-direction: column;
        img {
          width: 225px;
          height: 170px;
        }
      }
    }
  `;
});

export const ImageContainerBox = styled(Box, {
  name: "ImageContainerBox",
  shouldForwardProp,
})(({ theme }) => {
  return css`
    position: relative;

    .${typographyClasses.body2} {
      background-color: rgba(0, 0, 0, 0.7);
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 2;
      color: ${theme.palette.colors.white};
      font-size: 12px;
      font-style: normal;
      font-weight: 400;
      line-height: 11.72px;
      border-radius: 2px;
      padding: 2px 4px;
      position: absolute;
      right: 10px;
      bottom: 10px;
      z-index: 9999;
    }
  `;
});
