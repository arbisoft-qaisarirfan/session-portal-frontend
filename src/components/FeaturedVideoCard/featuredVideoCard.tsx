import React, { FC } from "react";

import Box from "@mui/material/Box";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import Image from "next/image";

import { formatDateTime, trimTextLength } from "@/utils/utils";

import { FeaturedVideoCardContainer } from "./styled";
import { FeaturedVideoCardProps } from "./types";

const FeaturedVideoCard: FC<FeaturedVideoCardProps> = ({
  className,
  event_time,
  thumbnail,
  workstream_id,
  title,
  description,
  isVisible = false,
  width = "100%",
}) => {
  const inAppThumbnail = `http://localhost:8000/${thumbnail}`;

  return isVisible ? (
    <FeaturedVideoCardContainer $width={width} className={className} data-testid="video-card">
      <CardContent>
        <Image alt={title} height={192} width={315} src={thumbnail ? inAppThumbnail : "/assets/images/temp-youtube-logo.webp"} />
        <Box className="video-detail">
          <Typography variant="h3" component="div" title={title}>
            {trimTextLength(title, 70)}
          </Typography>
          <Typography variant="body2" className="organizer-name" data-testid="video-card-organizer">
            {workstream_id}
          </Typography>
          <Typography variant="body2" className="date-time" data-testid="video-card-date-time">
            {formatDateTime(event_time)}
          </Typography>
          <Typography variant="body2" className="video-description" data-testid="video-description">
            {trimTextLength(description, 200)}
          </Typography>
        </Box>
      </CardContent>
    </FeaturedVideoCardContainer>
  ) : null;
};

export default FeaturedVideoCard;
