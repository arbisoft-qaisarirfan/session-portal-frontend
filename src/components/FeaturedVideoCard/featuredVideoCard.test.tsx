import userEvent from "@testing-library/user-event";

import { customRender as render, screen } from "@/jest/utils/testUtils";

import FeaturedVideoCard from "./featuredVideoCard";
import { FeaturedVideoCardProps } from "./types";

describe("FeaturedVideoCard", () => {
  const mockProps: FeaturedVideoCardProps = {
    className: "custom-class",
    event_time: "Jan 01, 2024",
    thumbnail: "https://example.com/sample.jpg",
    title: "Sample Video Title",
    workstream_id: "Sample Video Organizer",
    description: "Sample video description",
  };

  test("should render the component with provided props", () => {
    render(<FeaturedVideoCard {...mockProps} />);
    expect(screen.getByText(mockProps.title)).toBeInTheDocument();
    expect(screen.getByTestId("video-card-date-time")).toHaveTextContent(mockProps.event_time);
    expect(screen.getByTestId("video-card-organizer")).toHaveTextContent(mockProps.workstream_id);
    expect(screen.getByTestId("video-description")).toHaveTextContent(mockProps.description);
    const imgUrl = screen.getByRole("img", { name: mockProps.title }).getAttribute("src") ?? "";
    expect(decodeURIComponent(imgUrl)).toContain(mockProps.thumbnail);
  });

  test("should display default image when imgUrl is not provided", () => {
    render(<FeaturedVideoCard {...mockProps} thumbnail={undefined} />);
    const imgUrl = screen.getByRole("img", { name: mockProps.title }).getAttribute("src") ?? "";
    expect(decodeURIComponent(imgUrl)).toContain("/assets/images/temp-youtube-logo.webp");
  });

  test("should render the organizer name", () => {
    render(<FeaturedVideoCard {...mockProps} />);
    expect(screen.getByText(mockProps.workstream_id)).toBeInTheDocument();
    expect(screen.getByTestId("video-card-organizer")).toBeInTheDocument();
  });

  test("should not render when isVisible is false", () => {
    render(<FeaturedVideoCard {...mockProps} isVisible={false} />);
    expect(screen.queryByTestId("video-card")).not.toBeInTheDocument();
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<FeaturedVideoCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render correctly with different width values", () => {
    const { container } = render(<FeaturedVideoCard {...mockProps} width="200px" />);
    const card = container.firstChild as HTMLElement;
    expect(card).toHaveStyle("width: 200px");
  });

  test("should not render video description when empty", () => {
    const mockPropsEmptyDescription = { ...mockProps, description: "" };
    render(<FeaturedVideoCard {...mockPropsEmptyDescription} />);

    expect(screen.queryByTestId("video-description")).not.toBeInTheDocument();
  });

  test("should have the correct alt text for the image", () => {
    render(<FeaturedVideoCard {...mockProps} />);
    const image = screen.getByRole("img", { name: mockProps.title });
    expect(image).toHaveAttribute("alt", mockProps.title);
  });

  test("should render empty date when date is not provided", () => {
    const mockPropsEmptyDate = { ...mockProps, date: "" };
    render(<FeaturedVideoCard {...mockPropsEmptyDate} />);
    expect(screen.getByTestId("video-card-date-time")).toHaveTextContent("");
  });

  test("should render the default class name", () => {
    render(<FeaturedVideoCard {...mockProps} />);
    expect(screen.getByTestId("video-card")).toHaveClass("custom-class");
  });

  test("should truncate title if it is too long", () => {
    const longTitle = "A very long video title that should be truncated properly in the UI";
    render(<FeaturedVideoCard {...mockProps} title={longTitle} />);

    expect(screen.getByText(/A very long video title/i)).toBeInTheDocument();
  });

  test("should handle special characters in title", () => {
    // eslint-disable-next-line quotes
    const specialTitle = 'Video & Special "Title"!';
    render(<FeaturedVideoCard {...mockProps} title={specialTitle} />);

    expect(screen.getByText(specialTitle)).toBeInTheDocument();
  });

  test("should apply custom class name", () => {
    render(<FeaturedVideoCard {...mockProps} className="custom-class" />);
    expect(screen.getByTestId("video-card")).toHaveClass("custom-class");
  });

  test("should allow user interactions on the card", async () => {
    render(<FeaturedVideoCard {...mockProps} />);
    const user = userEvent.setup();
    const videoCard = screen.getByTestId("video-card");

    await user.click(videoCard);
    expect(videoCard).toBeInTheDocument(); // Ensures no crash on click
  });
});
