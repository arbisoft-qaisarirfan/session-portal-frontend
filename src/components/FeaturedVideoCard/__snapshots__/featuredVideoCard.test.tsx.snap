// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedVideoCard should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root custom-class css-jwtqct-MuiPaper-root-MuiCard-root-FeaturedVideoCardContainer-root"
    data-testid="video-card"
    style="--Paper-shadow: var(--mui-shadows-1); --Paper-overlay: var(--mui-overlays-1);"
  >
    <div
      class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-15l7p1e-ImageContainerBox-root"
      >
        <img
          alt="Sample Video Title"
          data-nimg="1"
          decoding="async"
          height="192"
          loading="lazy"
          src="/_next/image?url=http%3A%2F%2Flocalhost%3A1234%2Fassets%2Fimages%2Ftemp-youtube-logo.webp&w=640&q=75"
          srcset="/_next/image?url=http%3A%2F%2Flocalhost%3A1234%2Fassets%2Fimages%2Ftemp-youtube-logo.webp&w=384&q=75 1x, /_next/image?url=http%3A%2F%2Flocalhost%3A1234%2Fassets%2Fimages%2Ftemp-youtube-logo.webp&w=640&q=75 2x"
          style="color: transparent;"
          width="315"
        />
        <div
          class="MuiTypography-root MuiTypography-bodyMedium video-duration css-149i8z5-MuiTypography-root"
          data-testid="video-duration"
        >
          30:00
        </div>
      </div>
      <div
        class="video-detail MuiBox-root css-0"
      >
        <div
          class="MuiTypography-root MuiTypography-h3 css-1fho9p8-MuiTypography-root"
          title="Sample Video Title"
        >
          Sample Video Title
        </div>
        <p
          class="MuiTypography-root MuiTypography-bodyMedium organizer-name css-149i8z5-MuiTypography-root"
          data-testid="video-card-organizer"
        >
          Sample Video Organizer
        </p>
        <p
          class="MuiTypography-root MuiTypography-bodySmall date-time css-1pot96k-MuiTypography-root"
          data-testid="video-card-date-time"
        >
          Oct 22, 2024
        </p>
        <p
          class="MuiTypography-root MuiTypography-bodySmall video-description css-1pot96k-MuiTypography-root"
          data-testid="video-description"
        >
          Sample video description
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
